data-prepper-core/src/main/java/org/opensearch/dataprepper/peerforwarder/RemotePeerForwarder.java

```diff
@@ -141,8 +141,17 @@ private Map<String, List<Record<Event>>> groupRecordsBasedOnIdentificationKeys(
             final Event event = record.getData();
 
             final List<String> identificationKeyValues = new LinkedList<>();
+            boolean identificationKeysNotFound = false;
             for (final String identificationKey : identificationKeys) {
-                identificationKeyValues.add(event.get(identificationKey, Object.class).toString());
+                Object identificationKeyValue = event.get(identificationKey, Object.class);
+                if (identificationKeyValue == null) {
```

Comment:
Is this customer error ? Is this important enough to log and metric when this happens ?